src/HDF5.jl

```diff
@@ -709,6 +709,16 @@ function h5read(filename, name::String, indices::Tuple{Vararg{Union{AbstractRang
     dat
 end
 
+function h5read(filename, name::String, cr::CartesianRange)
```

Comment:
@timholy  it seems that ` cr::CartesianRange` is sufficient in the function signature, is the  `cr::CartesianRange{CartesianIndex{N}}) where {N}...` restriction needed  (which actually doesn't work on 0.7)?